### PR TITLE
Add the specification for the I/O library

### PR DESCRIPTION
**Pull request by BuddhiWathsala**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/2254

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+public isolated function fileReadJson(string path) returns json|Error;
+```
+
+The following API writes given JSON to a given file.
+
+```ballerina
+# Write a JSON to a file.
+# ```ballerina
+# json content = {"name": "Anne", "age": 30};
+# io:Error? content = io:fileWriteJson("./resources/myfile.json");
```
Comment (daneshk) [inline docs/spec/spec.md:354]:
```suggestion
# io:Error? result = io:fileWriteJson("./resources/myfile.json", content);
```

Comment (BuddhiWathsala) [reply docs/spec/spec.md]: Fixed with https://github.com/ballerina-platform/module-ballerina-io/pull/434/commits/17965a6c228972db917890880b504982b22f1799

**Review by daneshk — APPROVED** [review]
LGTM

File: docs/spec/spec.md [hunk 2]
```diff
@@ -0,0 +1,423 @@
+# Specification: Ballerina Time Library
```
Comment (pubudu91) [inline docs/spec/spec.md:1]: IO Library?

**Pull request merged**